### ISSUE TITLE
amagama-live request failed --> unknown error (Unbekannter Fehler) 

### ISSUE THREAD

**Issue by ChrSchwaer** [translate/pootle#5275]
We use qt translation files (.ts) for translation
For Example File de_DE.ts:

```xml
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1" language="de_DE" sourcelanguage="en_GB">
```

When we switch to the next Translation in Browser we always get the Error Message (in German: "Unbekannter Fehler") in English maybe: "Unknown Error"

After using the Browser Developer Tools we find out that it will send this message:
_https://amagama-live.translatehouse.org/api/v1/en_GB/de/unit/?source=Version%3A&jsoncallback=_jsonp275045_

The Answer for this request is  :collision: :

```xml
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server.  If you entered the URL manually please check your spelling and try again.</p>
```

We find out, if we change the request to (just in developer tool):
_https://amagama-live.translatehouse.org/api/v1/en/de/unit/?source=Version%3A&jsoncallback=_jsonp275045_

we get a correct answer!
The diffrent ist here:
_...amagama-live.translatehouse.org/api/v1/en**_GB**/de/unit/?source=Version%3A&jsoncallback=_jsonp275045_
__________________________________________________^
If we just set here **"en"** instead of **"en_GB"** it works!

How, or were this will be set! It looks like you set the "sourcelanguage" from .ts file but you just need a part of it!

What can we do that we didn't get the Error Message?

**Comment by dwaynebailey** [comment]
@ChrSchwaer thanks for the report.  Pootle should handle the error better ito not failing when it doesn't find a source language.

But amagama does not host any `en_GB` source text, only `en`/`en_US`.

Is your source language genuinely en_GB? Or is it simply en/en_US.  The quickest fix would be to change the source language in your .ts files.

It's a much longer fix on our side to get amagama to decide to normalise the searches as sometimes you wouldn't want to search for the language only domain when you have a more defined search.

**Comment by ChrSchwaer** [comment]
@dwaynebailey thanx for your quick answer.
I tested diffrent settings in the .ts file. Actualy the sourcelanguage entry in the .ts file doesn't matter. The important thing is to set the right sourcelanguage in the pootle project settings!
There you have to choose "English - en" if you there set for example "English (UK) - en_GB" you got this error!

✔️  Result for me: I set the Sourcelanguage in pootle project Setting just to to "English - en" and now it works! 😄

**Comment by unho** [comment]
Closing as resolved.

**Comment by bourdeau** [comment]
Hello,

I've the exact same problem on my fresh Pootle install **(Pootle 2.7.6 (Django 1.7.11, Translate Toolkit 1.13.0))**

In my project settings I couldn't find French so I set it to English - en and changed my **xliff** to en_GB:

```xml
<?xml version="1.0" encoding="utf-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:jms="urn:jms:translation" version="1.2">
  <file date="2016-12-19T17:28:58Z" source-language="en_GB" target-language="en_GB" datatype="plaintext" original="not.available">
[...]
```
the I ran **pootle update_stores** but I had the same error. I tried with en as source-language and ran in the same troubles.

What am I doing wrong here? (except the target-language which is absurd...)
